### PR TITLE
Create http4s mAuth middleware

### PR DESCRIPTION
**Pull request by fserra-mdsol**

### CODE REVIEW

File: modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+      .assert
+  }
+
+  test("reject if public key cannot be found") {
+    val localClient = new ClientPublicKeyProvider {
+      override def getPublicKey(appUUID: UUID): Future[Option[PublicKey]] =
+        if (appUUID == appUuid) {
+          Future(none)
+        } else Future.failed(new Throwable("Wrong app UUID"))
+      override def getPublicKeyIO(appUUID: UUID): IO[Option[PublicKey]] = ???
```
Comment (jatcwang) [inline modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala:147]: Can you remove this from the public interface while you're at it? :) I don't think it's suppose to be public

Comment (fserra-mdsol) [reply modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala:147]: Yep, on it!

**Review by jatcwang — APPROVED** [review]

**Review by jatcwang — APPROVED** [review]

File: modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala (hunk 1 above)
Comment (tmilner) [reply modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala]: I think we did it on purpose so you dont need to use future in places like this. The future API is just calling this with an .unsafeToFuture.

**Review by tmilner — APPROVED** [review]

File: modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala (hunk 1 above)
Comment (tmilner) [reply modules/mauth-authenticator-http4s/src/test/scala/com/mdsol/mauth/http4s/MAuthMiddlewareSuite.scala]: If its not used anywhere though its fine to remove it.

**Pull request merged**